<?xml version="1.0"?>
<launch>
    <!-- MAVROS posix SITL environment launch script -->
    <!-- launches Gazebo environment and 2x: MAVROS, PX4 SITL, and spawns vehicle -->
    <!-- vehicle model and world -->
    <arg name="est" default="ekf2"/>
    <arg name="vehicle" default="iris"/>
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>
    <arg name="UAV" default="UAV1"/>

    <!-- UAV1 -->
    <group ns="$(arg UAV)">
        <!-- MAVROS and vehicle configs -->
        <arg name="ID" value="0"/>
        <arg name="fcu_url" default="udp://:14540@localhost:14580"/>
        <!-- PX4 SITL and vehicle spawn -->
        <include file="$(find px4)/launch/single_vehicle_spawn_sdf.launch">
            <arg name="x" value="$(arg x)"/>
            <arg name="y" value="$(arg y)"/>
            <arg name="z" value="$(arg z)"/>
            <arg name="R" value="0"/>
            <arg name="P" value="0"/>
            <arg name="Y" value="0"/>
            <arg name="vehicle" value="$(arg vehicle)"/>
            <arg name="mavlink_udp_port" value="14560"/>
            <arg name="mavlink_tcp_port" value="4560"/>
            <arg name="ID" value="$(arg ID)"/>
        </include>

        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="$(eval 1 + arg('ID'))"/>
            <arg name="tgt_component" value="1"/>
        </include>

        <!-- Xbox control -->
        <!-- <param name="joy_node/dev" value="/dev/input/js0"/>
        <node name="joy_node" pkg="joy" type="joy_node"/>
        <node name="xboxVelControl" pkg="quadnodes" type="xboxVelControl"/> -->

        <node name="tfWorldUAV" pkg="quadnodes" type="tfWorldUAV" output="screen">
          <param name="x_offset" value="$(arg x)"/>
          <param name="y_offset" value="$(arg y)"/>
          <param name="z_offset" value="$(arg z)"/>
          <param name="UAV_Name" value="$(arg UAV)"/>
        </node>

        <node name="copter_visualization" pkg="mavros_extras" type="visualization">
            <param name="num_rotors" value="4"/>
            <param name="arm_len" value="0.22"/>
            <!-- <param name="UAV_Name" value="$(arg UAV)"/> -->
            <param name="fixed_frame_id" value="worldUAVs"/>
            <param name="child_frame_id" value="$(arg UAV)"/>
        </node>

    </group>
</launch>
<!-- to add more UAVs (up to 10):
Increase the id
Change the name space
Set the FCU to default="udp://:14540+id@localhost:14550+id"
Set the malink_udp_port to 14560+id) -->
